### PR TITLE
Change the default timeout for all backup jobs

### PR DESCRIPTION
**Pull request by sivanov-nuodb**
**Issue**
Hotcopy timeout value for backup cron-jobs can lead to false positives as the underlying hot-copy operation continues to execute in the background on the SM but the job can fail due to timeout leaving a full backup set not recorded in the KV store.

**Issue**
Journal hotcopy timeout setting is not used for journal backup jobs.

**Changes**
- Default timeout for `full`, `incremental` and `journal` backup jobs has been changed to `0` which means that they should wait forever for the underlying hot-copy operation to complete.
- Journal backup timeout is not used when invoking `nuobackup`
- Make `nuobackup` non-blocking. Similar to our other commands, the user needs to specify `--timeout` if he wants it to block synchronously.

**Product changes**
Starting with NuoDB 4.2.1+ `nuodocker backup database` will use client-side polling to wait for the hot-copy operation to finish.

**Testing**
- integration tests
- regression testing

### CODE REVIEW

File: test/integration/template_backup_test.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func TestDatabaseBackupCronJobDisabled(t *testing.T) {
 	// Run RenderTemplate to render the template and capture the output.
 	_, err := helm.RenderTemplateE(t, options, helmChartPath, "release-name", []string{"templates/cronjob.yaml"})
 
-	assert.Contains(t, err, "could not find template")
+	assert.Contains(t, err.Error(), "could not find template")
```
Comment (mkysel) [inline test/integration/template_backup_test.go:41]: should this just be `assert.ErrorContains`?

**Review by mkysel — APPROVED** [review]

File: test/integration/template_backup_test.go (hunk 1 above)
Comment (sivanov-nuodb) [reply test/integration/template_backup_test.go:41]: There is no `ErrorContains` in github.com/stretchr/testify/assert. But I needed it for `Subset` function.

Comment (mkysel) [reply test/integration/template_backup_test.go:41]: I see that you changed gotest to stretcher. Got it!

I wonder why it was left behind...

File: stable/database/templates/cronjob.yaml [hunk 2]
```diff
@@ -54,7 +54,6 @@ spec:
             - { name: DB_NAME,             value: "{{ .Values.database.name }}"     }
             - { name: NUOCMD_API_SERVER,   value: "{{ template "admin.address" . }}:8888" }
             - { name: PEER_ADDRESS,        value: "{{ template "admin.address" . }}" }
-            - { name: TIMEOUT,             value: "{{ .Values.database.sm.hotCopy.timeout }}" }
```
Comment (sivanov-nuodb) [inline stable/database/templates/cronjob.yaml:57]: @NikTJ777 Do you remember why these `TIMEOUT` env vars are set?

File: test/integration/template_backup_test.go (hunk 1 above)
Comment (mkysel) [reply test/integration/template_backup_test.go:41]: could you remove it from `go.mod` and `go.sum` so that people don't use it accidentally?

Comment (sivanov-nuodb) [reply test/integration/template_backup_test.go:41]: Good point! I will remove `gotest.tools` ... it looks like it isn't used anywhere else.

**Review by adriansuarez — APPROVED** [review]

**Pull request merged**